### PR TITLE
Omitt Jacobian calculation for scaling issues

### PR DESCRIPTION
**Pull request by bernhardbachmann**
### Related Issues

#10287 

### Purpose

Improve Jacobian calculation in GBODE (only update, if necessary). By default the Jacobian has been calculated each time a/the nonlinear system has to be solved. This happened at least at each integration time step. 

### Approach

Reuse the already calculated Jacobian stored in the kinsol data.

### CODE REVIEW

File: OMCompiler/SimulationRuntime/c/simulation/solver/sundials_util.c [hunk 1]
```diff
@@ -78,6 +78,36 @@ void setJacElementSundialsSparse(int row, int column, int nth, double value, voi
   SM_DATA_S(A)[nth] = value;
 }
 
+/**
+ * @brief             Scaling of a sparse matrix column-wise by a vector
+ *
+ * @param A           Sparse matrix in CSC.
+ * @param vScale      Vector for scaling
+ * @return *SUNMatrix Returns the scaled sparse matrix
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/sundials_util.c:86]: This is not a pointer

```suggestion
 * @return SUNMatrix  Returns the scaled sparse matrix
```

**Review by AnHeuermann — APPROVED** [review]

**Comment by AnHeuermann** [comment]
Let's see how much chaos errupts over the library tests during the weekend when this gets merged in a few minutes.

**Review by casella — APPROVED** [review]
LGTM. Why should it generate chaos?

**Comment by AnHeuermann** [comment]
No specific reason, but changes to the non-linear solvers tend to change more than we anticipate.
But in this case I can't see any changes in the [regression reports](https://libraries.openmodelica.org/branches/history/master/2023-03-02%2022:43:15..2023-03-04%2000:08:48.html) that seem to be a result of this PR.
Maybe [https://libraries.openmodelica.org/branches/master/ScalableTestGrids_noopt/files/ScalableTestGrids_noopt_ScalableTestGrids.Models.Type1_N_4_M_4.sim](Type1_N_4_M_4) or it changes the status randomly.

**Comment by casella** [comment]
> No specific reason, but changes to the non-linear solvers tend to change more than we anticipate. But in this case I can't see any changes in the [regression reports](https://libraries.openmodelica.org/branches/history/master/2023-03-02%2022:43:15..2023-03-04%2000:08:48.html) that seem to be a result of this PR.

LGTM

> Maybe [https://libraries.openmodelica.org/branches/master/ScalableTestGrids_noopt/files/ScalableTestGrids_noopt_ScalableTestGrids.Models.Type1_N_4_M_4.sim](Type1_N_4_M_4) or it changes the status randomly.

I'll check them ASAP, but I guess this commit cannot be the culprit, since we are not using GBODE for that.

**Pull request merged**